splice_machine/src/main/java/com/splicemachine/derby/impl/storage/SpliceRegionAdmin.java

```diff
@@ -89,6 +92,290 @@
+        }
+        catch (Throwable t) {
+            resultSets[0] = ProcedureUtils.generateResult("Error", t.getLocalizedMessage());
+        }
+    }
+
+    /**
+     *
+     * @param schemaName
+     * @param tableName
+     * @param resultSets
+     */
+    public static void LOCALIZE_INDEXES_FOR_TABLE(String schemaName, String tableName, ResultSet[] resultSets) throws Exception {
+
+        Connection conn = SpliceAdmin.getDefaultConn();
+        LanguageConnectionContext lcc = conn.unwrap(EmbedConnection.class).getLanguageConnection();
+        TransactionController tc = lcc.getTransactionExecute();
+        TxnView txn = ((SpliceTransactionManager) tc).getActiveStateTxn();
+        schemaName = EngineUtils.validateSchema(schemaName);
+        tableName = EngineUtils.validateTable(tableName);
+        long conglomerateNumber = getConglomerateNumber(lcc, schemaName, tableName);
+        PartitionFactory partitionFactory = SIDriver.driver().getTableFactory();
+        Partition table = partitionFactory.getTable(Long.toString(conglomerateNumber));
+        List<Partition> partitions = table.subPartitions(new byte[0], new byte[0], true);
+        List<String> messages = Lists.newArrayList();
+        if (partitions.size() > 1) {
+            messages.add("Operation not supported for table with more than 1 region.");
+        } else {
+            MoveRegionTask task = new MoveRegionTask(txn, schemaName, tableName);
+            messages.addAll(task.call());
+        }
+        resultSets[0] = processResult(lcc, conn, messages);
+    }
+
+    /**
+     *
+     * @param schemaName
+     * @param resultSets
+     * @throws Exception
+     */
+    public static void LOCALIZE_INDEXES_FOR_SCHEMA(String schemaName, ResultSet[] resultSets) throws Exception {
+
+        List<String> tables = Lists.newArrayList();
+        schemaName = EngineUtils.validateSchema(schemaName);
+        String sql = String.format("select t.tablename from sysvw.systablesview t, sysvw.sysschemasview s, sysvw.sysconglomeratesview c" +
+                " where s.schemaname='%s' and s.schemaid=t.schemaid and c.tableid=t.tableid and stored is null and not c.isindex", schemaName);
+        Connection connection = SpliceAdmin.getDefaultConn();
+        try (Statement s = connection.createStatement();
+             ResultSet rs = s.executeQuery(sql)) {
+            while (rs.next()) {
+                tables.add(rs.getString(1));
+            }
+        }
+        LanguageConnectionContext lcc = connection.unwrap(EmbedConnection.class).getLanguageConnection();
+        TransactionController tc = lcc.getTransactionExecute();
+        TxnView txn = ((SpliceTransactionManager) tc).getActiveStateTxn();
+
+        List<MoveRegionTask> callables = Lists.newArrayList();
+        for (String table : tables) {
+            callables.add(new MoveRegionTask(txn, schemaName, table));
```

Comment:
should MoveRegionTask be called only for tables with 1 region, such as in LOCALIZE_INDEXES_FOR_TABLE?
with additional condition
`if (partitions.size() > 1) {`

Reply:
This needs to get metadata for each table serially. With thousands tables in one schema, this can be slow. MoveRegionTask ignores tables with more than 1 region.